lms/static/scripts/frontend_apps/components/test/GradingCommentButton-test.js

```diff
@@ -0,0 +1,148 @@
+import { mount } from 'enzyme';
+
+import { checkAccessibility } from '../../../test-util/accessibility';
+import GradingCommentButton from '../GradingCommentButton';
+
+const noop = () => {};
+
+describe('GradingCommentButton', () => {
+  const getToggleButton = wrapper =>
+    wrapper.find('Button[data-testid="comment-toggle-button"]');
+
+  const togglePopover = wrapper =>
+    getToggleButton(wrapper).find('button').simulate('click');
+
+  const commentPopoverExists = wrapper =>
+    wrapper.exists('[data-testid="comment-popover"]');
+
+  const renderComponent = (props = {}) =>
+    mount(
+      <GradingCommentButton
+        comment="Good job!"
+        disabled={false}
+        loading={false}
+        onSubmit={noop}
+        onInput={noop}
+        {...props}
+      />
+    );
+
+  it('allows comment popover to be toggled', () => {
+    const wrapper = renderComponent();
+
+    // Popover is initially hidden
+    assert.isFalse(commentPopoverExists(wrapper));
+    assert.isFalse(getToggleButton(wrapper).prop('expanded'));
+
+    // Clicking the toggle will display the popover
+    togglePopover(wrapper);
+    assert.isTrue(commentPopoverExists(wrapper));
+    assert.isTrue(getToggleButton(wrapper).prop('expanded'));
+
+    // A second click will hide the popover
+    togglePopover(wrapper);
+    assert.isFalse(commentPopoverExists(wrapper));
+    assert.isFalse(getToggleButton(wrapper).prop('expanded'));
+  });
+
+  it('hides the popover when `Escape` is pressed', () => {
+    const wrapper = renderComponent();
+
+    // Show popover
+    togglePopover(wrapper);
+    assert.isTrue(commentPopoverExists(wrapper));
+
+    document.body.dispatchEvent(
+      new KeyboardEvent('keydown', { key: 'Escape' })
+    );
+    wrapper.update();
+
+    assert.isFalse(commentPopoverExists(wrapper));
+  });
+
+  it('hides the popover when clicking away', () => {
+    const wrapper = renderComponent();
+
+    // Show popover
+    togglePopover(wrapper);
+    assert.isTrue(commentPopoverExists(wrapper));
+
+    const externalButton = document.createElement('button');
+    document.body.append(externalButton);
+    externalButton.click();
+    wrapper.update();
+
+    assert.isFalse(commentPopoverExists(wrapper));
+
+    externalButton.remove();
```

Comment:
Might want to wrap this in `try { ... } finally { ... }` so the DOM is cleaned up even if the assert were to fail, and avoid potentially affecting other tests.